src/lib/structures/InviteStore.ts

```diff
@@ -0,0 +1,54 @@
+import Collection from '@discordjs/collection';
+import { resolveOnErrorCodes } from '@utils/util';
+import { api } from '@utils/Models/Api';
+import { APIErrors, Time } from '@utils/constants';
+import { APIInviteData } from '@lib/types/DiscordAPI';
+import { Client } from 'discord.js';
+
+export class InviteStore extends Collection<string, InviteCodeEntry> {
+
+	private readonly client: Client;
+
+	public constructor(client: Client) {
+		super();
+		this.client = client;
+		this.client.setInterval(() => {
+			const deleteAt = Date.now() - (Time.Minute * 15);
+			this.sweep(value => value.fetchedAt < deleteAt);
+		}, Time.Minute);
+	}
+
+	public async fetch(code: string) {
+		const previous = this.get(code);
+		if (typeof previous !== 'undefined') return previous;
+
+		const data = await resolveOnErrorCodes(api(this.client).invites(code).get(), APIErrors.UnknownInvite) as APIInviteData | null;
+		if (data === null) {
+			const resolved: InviteCodeEntry = { valid: false, fetchedAt: Date.now() };
+			this.set(code, resolved);
+			return resolved;
+		}
+
+		const resolved: InviteCodeEntry = {
+			valid: true,
+			guildID: 'guild' in data ? data.guild.id : null,
```

Comment:
```suggestion
			guildID: Reflect.has(data, 'guild') ? data.guild.id : null,
```

![image](https://user-images.githubusercontent.com/4019718/78461804-5afa9900-76cc-11ea-974f-029272e9ca54.png)

Reply:
Yeah but it provides no assertions or whatsoever, therefore, your code change would cause a compile-time error.

Another option is to do `data.guild!.id` in the ternary with the reflect.